optimizer/plan/cost.go

```diff
@@ -87,6 +81,41 @@ func (c *costEstimator) Leave(p Plan) (Plan, bool) {
 	}
 	return p, true
 }
+
+func (c *costEstimator) tableScan(v *TableScan) {
+	var rowCount float64
+	if len(v.Ranges) == 1 && v.Ranges[0].LowVal == math.MinInt64 && v.Ranges[0].HighVal == math.MaxInt64 {
+		// full range use default row count.
+		rowCount = FullRangeCount
+	} else {
+		for _, v := range v.Ranges {
+			// for condition like 'a = 0'.
+			if v.LowVal == v.HighVal {
+				rowCount++
+				continue
+			}
+			// For condition like 'a < 0'.
+			if v.LowVal == math.MinInt64 {
+				rowCount += HalfRangeCount
+			}
+			// For condition like 'a > 0'.
+			if v.HighVal == math.MaxInt64 {
+				rowCount += HalfRangeCount
+			}
+			// For condition like 'a > 0 and a < 1'.
+			rowCount += 100
+		}
+	}
+	v.startupCost = 0
+	if v.limit == 0 {
+		// limit is zero means no limit.
```

Comment:
This can be optimized later, when we find 0 limit, we don't need to optimize it, but currently it is used to estimate cost, it doesn't affect the correctness of the execution.